src/Autofac/Core/Resolving/Pipeline/ResolveRequestContext.cs

```diff
@@ -23,40 +23,107 @@
             ISharingLifetimeScope scope,
             DiagnosticListener diagnosticSource)
-            : base(owningOperation, request, scope, diagnosticSource)
         {
+            Operation = owningOperation;
+            ActivationScope = scope;
+            Parameters = request.Parameters;
+            _resolveRequest = request ?? throw new ArgumentNullException(nameof(request));
+            PhaseReached = PipelinePhase.ResolveRequestStart;
+            DiagnosticSource = diagnosticSource;
         }
 
-        /// <summary>
-        /// Completes the request context; invokes any event handlers.
-        /// </summary>
-        public void Complete()
+        /// <inheritdoc />
+        public ResolveOperationBase Operation { get; }
+
+        /// <inheritdoc />
+        public ISharingLifetimeScope ActivationScope { get; private set; }
+
+        /// <inheritdoc />
+        public IComponentRegistration Registration => _resolveRequest.Registration;
+
+        /// <inheritdoc />
+        public Service Service => _resolveRequest.Service;
+
+        /// <inheritdoc />
+        public IComponentRegistration? DecoratorTarget => _resolveRequest.DecoratorTarget;
+
+        /// <inheritdoc />
+        [DisallowNull]
+        public object? Instance
+        {
+            get => _instance;
+            set => _instance = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        /// <inheritdoc />
+        public bool NewInstanceActivated => Instance is object && PhaseReached == PipelinePhase.Activation;
+
+        /// <inheritdoc />
+        public DiagnosticListener DiagnosticSource { get; }
+
+        /// <inheritdoc />
+        public IEnumerable<Parameter> Parameters { get; private set; }
+
+        /// <inheritdoc />
+        public PipelinePhase PhaseReached { get; set; }
+
+        /// <inheritdoc />
+        public IComponentRegistry ComponentRegistry => ActivationScope.ComponentRegistry;
+
+        /// <inheritdoc />
+        public event EventHandler<ResolveRequestCompletingEventArgs>? RequestCompleting;
+
+        /// <inheritdoc />
+        public DecoratorContext? DecoratorContext { get; set; }
+
+        /// <inheritdoc />
+        public void ChangeScope(ISharingLifetimeScope newScope) =>
+            ActivationScope = newScope ?? throw new ArgumentNullException(nameof(newScope));
+
+        /// <inheritdoc />
+        public void ChangeParameters(IEnumerable<Parameter> newParameters) =>
+            Parameters = newParameters ?? throw new ArgumentNullException(nameof(newParameters));
+
+        /// <inheritdoc />
+        public object ResolveComponent(ResolveRequest request) =>
+            Operation.GetOrCreateInstance(ActivationScope, request);
+
+
+        /// <inheritdoc/>
+        public void CompleteRequest()
         {
-            // Let the base class raise events.
-            CompleteRequest();
+            EventHandler<ResolveRequestCompletingEventArgs>? handler = RequestCompleting;
```

Comment:
Is there a reason, why this is copied to a local variable again? Why not just `RequestCompleting?.Invoke(this, new ...)`;

Reply:
Something in my brain is telling me that it's possible for that RequestCompleting handler to be changed from inside the event handler itself? Maybe? Or it's a GC thing? I can't entirely remember tbh.